R/generate_variability_terms.R

```diff
@@ -131,9 +133,17 @@ generate_ruv <- function(
   if (!is.null(seed)) {
     set.seed(seed)
   }
-  ruv$prop <- 1 + rnorm(nrow(ruv), 0, prop)
-  ruv$add <- rnorm(nrow(ruv), 0, add)
+  if (ltbs) {
+    # data is returned from `PKPDsim::sim` in the linear domain already so
+    # apply additive error (log domain) as proportional error (linear domain).
+    ruv$prop <- rnorm(nrow(ruv), 1, PKPDsim::add_ruv(1, ruv = list(prop = add, add = 0)))
+    ruv$add <- 0
+  } else {
+    ruv$prop <- rnorm(nrow(ruv), 1, PKPDsim::add_ruv(1, ruv = list(prop = prop, add = 0)))
+    ruv$add <- rnorm(nrow(ruv), 0.00001, PKPDsim::add_ruv(0, ruv = list(prop = 0.00001, add = add)))
```

Comment:
I think this is a case where using PKPDsim::add_ruv makes the code harder to follow and more error prone. (for example, the additive error here scales proportionally, even if just by a little)

@roninsightrx what do you think? is this more of a function re-write to make use of this PKPDsim tool? or should we leave it as it was without calling the PKPDsim function?

Reply:
Hmm, I'm not sure the above code is correct. Shouldn't this be just:
```
ruv$prop <- PKPDsim::add_ruv(rep(1, nrow(ruv)), ruv = list(prop = prop, add = 0))
ruv$add  <- PKPDsim::add_ruv(rep(1, nrow(ruv)), ruv = list(prop = 0, add = add))
```
Also, I don't think we should handle LTBS here, IMO it'd be more appropriate to do it at the spot where the error is added to the simulated data, then you can use the proper LTBS and not "proportional".

Optionally we can also add the `seed` argument to `add_ruv()`, I see we don't have it there yet.

Reply:
> Optionally we can also add the `seed` argument to `add_ruv()`, I see we don't have it there yet.

Actually, ignore that last suggestion about `seed`. Because of the sequential calling of add_ruv() that would actually induce 100% correlation between prop and add errors and thus inflate the error artificially.

Reply:
I had initially tried a similar code to what was suggested above and it was breaking tests centering additive error around 1; however, in adopting your suggestion and altering to (below), the tests pass. 
ruv$prop <- PKPDsim::add_ruv(rep(1, nrow(ruv)), ruv = list(prop = prop, add = 0))
ruv$add  <- PKPDsim::add_ruv(rep(0, nrow(ruv)), ruv = list(prop = 0, add = add))

My apologies if this suggestion is pulling in the wrong direction, it seemed the quick fix for my heparin project, but perhaps it is not the right move for the overall package.

Reply:
No worries. If this is getting too much into the weeds 'm also fine to keep the old code as-is, and make a ticket to fix at later point.